pacman/game.py

```diff
@@ -33,15 +33,43 @@ def add_ghost(self, agent):
             raise OutsideOfLegalPath("Cannot add agent\
                 to invalid board position")
 
-    def play_game(self, game_steps):
+    def check_rules(self, pacman_old_node, pacman_new_node,
+                    ghost_old_node, ghost_new_node):
+        collision = False
+        if (pacman_new_node.position == ghost_new_node.position):
+            collision = True
+        if ((pacman_new_node.position == ghost_old_node.position) and (
+                pacman_old_node.position == ghost_new_node.position)):
+            collision = True
+        return collision
```

Comment:
Wouldn't it be simpler and easier to understand to check for collision everytime the pacman agent moves ?
For instance something like (pseudo code):
```
for agent in agents:
    agent.move()
    if isinstance(agent, "PacMan"):
        for ghost in range(ghosts):
            if agent.current_node == ghost.current_node:
                end the game !
```